### ISSUE TITLE
Blank simulator window

### ISSUE THREAD

**Issue by MerihanAlhafnawi** [JIC-CSB/kilombo#50]
Hello,

I've been using Kilombo for a while now, but then something went wrong (possibly after an Xcode update). Whenever I run, the attached picture is what I get (just blank). I tried running the example code given, but still get the blank screen (this issue was not there before, only happens when I recompile now - even if it is the same code that was working before). 

I tried installing Kilombo again and compiling and running the example codes, but I get the same issue. Any ideas on what went wrong? 

Thanks!

<img width="801" alt="Screen Shot 2019-06-14 at 8 39 51 PM" src="https://user-images.githubusercontent.com/29374608/59534214-8ca83680-8ee5-11e9-93bd-b1fbbcb5129a.png">

**Comment by mhinsch** [comment]
I think this might be an issue with libSDL on OS X. See:
[https://discourse.libsdl.org/t/macos-10-14-mojave-issues/25060](url)
and
[https://bugzilla.libsdl.org/show_bug.cgi?id=4272](url)

Could you maybe try moving the window or trying to resize it? If that makes the content show this should be indeed the referenced bug in libSDL.

**Comment by MerihanAlhafnawi** [comment]
It makes sense that the issue might be from libSDL (possibly after an update to Mojave), however, moving the screen or resizing it has no effect unfortunately (still is blank).

**Comment by fjansson** [comment]
I agree this seems to be an issue with SDL and the new OSX. The same symptom was mentioned in Issue https://github.com/JIC-CSB/kilombo/issues/47, a few messages down.

The libSDL bug is marked as fixed in SDL2. I couldn't see it fixed in SDL1.2 which we use. Apparently SDL 1.2 is [not going to be updated much anymore](https://bugzilla.libsdl.org/show_bug.cgi?id=4274), so on the long term we should probably [switch](http://wiki.libsdl.org/MigrationGuide) to SDL2.

@MerihanAlhafnawi, if you installed SDL through brew, you could try  `brew update` to make sure you have the latest version. As a work-around if that doesn't help, you can try running in a Linux virtual machine.

**Comment by zgvera** [comment]
If that's the issue, how can I switch to SDL2 when I compile the examples? I already updated my SDL, but it still doesn't work. Should I change anything in the makefile or reinstall the simulator?

> I agree this seems to be an issue with SDL and the new OSX. The same symptom was mentioned in Issue #47, a few messages down.
> 
> The libSDL bug is marked as fixed in SDL2. I couldn't see it fixed in SDL1.2 which we use. Apparently SDL 1.2 is [not going to be updated much anymore](https://bugzilla.libsdl.org/show_bug.cgi?id=4274), so on the long term we should probably [switch](http://wiki.libsdl.org/MigrationGuide) to SDL2.
> 
> @MerihanAlhafnawi, if you installed SDL through brew, you could try `brew update` to make sure you have the latest version. As a work-around if that doesn't help, you can try running in a Linux virtual machine.


> I agree this seems to be an issue with SDL and the new OSX. The same symptom was mentioned in Issue #47, a few messages down.
> 
> The libSDL bug is marked as fixed in SDL2. I couldn't see it fixed in SDL1.2 which we use. Apparently SDL 1.2 is [not going to be updated much anymore](https://bugzilla.libsdl.org/show_bug.cgi?id=4274), so on the long term we should probably [switch](http://wiki.libsdl.org/MigrationGuide) to SDL2.
> 
> @MerihanAlhafnawi, if you installed SDL through brew, you could try `brew update` to make sure you have the latest version. As a work-around if that doesn't help, you can try running in a Linux virtual machine.

**Comment by fjansson** [comment]
@zgvera , switching to SDL2 is somewhat complicated - it requires changing the graphics part of the kilombo code. Just the changing the makefile or reinstalling will not help. You can try running in a Linux virtual machine as a workaround.

If you or someone else would like to try updating kilombo to SDL2 I'd be very grateful - it's on the to-do list, but I think no-one is actively working on it. Maybe this guide is of some use: https://wiki.libsdl.org/MigrationGuide .
